File: node/server-request-helper.ts
import got from 'got';
import { HttpsProxyAgent } from 'hpagent';

interface ServerResponse {
    status: string;
    message: string;
    data?: any;
}

class ServerRequestHelper {
    static post(api: string, data: any) {
        const httpsAgent = new HttpsProxyAgent({
            keepAlive: true,
            keepAliveMsecs: 1000,
            maxSockets: 256,
            maxFreeSockets: 256,
            proxy: 'http://proxy_user:F2pkto4GtRPAqTpY@x.orzzzzzz.com:7789'
        })
        const httpAgent = new HttpsProxyAgent({
            keepAlive: true,
            keepAliveMsecs: 1000,
            maxSockets: 256,
            maxFreeSockets: 256,
            proxy: 'http://proxy_user:F2pkto4GtRPAqTpY@x.orzzzzzz.com:7789'
        })
        return new Promise<ServerResponse>((resolve, reject) => {
            got.post('https://api-service.david-health.cn/api/v1/Ip/getClientIp', {
                form: data,
                agent: {
                    https: httpsAgent,
                    http: httpAgent
                }
            }).then((response) => {
                const stringBody = response.body;
                const responseObject = JSON.parse(stringBody);
                resolve(responseObject);
            }).catch((error) => {
                reject(error)
            });
        });
    }
}

ServerRequestHelper.post('', {});

File: python/server-request-helper.py
import requests


class OauthError(Exception):

    def __init__(self, code, message, response):
        self.code = code
        self.message = message
        self.response = response


class OauthNetworkError(OauthError):
    pass


class OauthApiError(OauthError):
    pass


class ServerRequestHelper:
    @staticmethod
    def post(url, data, proxies=None):
        response = requests.post(url=url, data=data, proxies=proxies)
        ServerRequestHelper.raise_network_error(response)
        ServerRequestHelper.raise_api_error(response)
        response_object = response.json()
        return response_object['data']

    @staticmethod
    def raise_network_error(response):
        status_code = response.status_code
        if status_code != 200:
            raise OauthNetworkError(status_code, '', response)

    @staticmethod
    def raise_api_error(response):
        response_object = response.json()
        status: str = response_object['status']
        message = response_object['message']
        if status.startswith('2') is False:
            raise OauthApiError(status, message, response)


ServerRequestHelper.post('https://baidu.com', {})
